src/Orleans/AssemblyLoader/AssemblyProcessor.cs

```diff
@@ -0,0 +1,170 @@
+﻿/*
+Project Orleans Cloud Service SDK ver. 1.0
+ 
+Copyright (c) Microsoft Corporation
+ 
+All rights reserved.
+ 
+MIT License
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and 
+associated documentation files (the ""Software""), to deal in the Software without restriction,
+including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+namespace Orleans.Runtime
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Reflection;
+
+    using Orleans.CodeGeneration;
+    using Orleans.Serialization;
+
+    /// <summary>
+    /// The assembly processor.
+    /// </summary>
+    internal class AssemblyProcessor
```

Comment:
I wonder as all the fields are statics and there's a locked `Initialize` method if there should be only single instance of these and some other pattern could be more recognizable? Such as some from http://csharpindepth.com/Articles/General/Singleton.aspx. Hmm, or some variant of those like
```
/// <summary>
    /// A generic singleton.
    /// </summary>
    public sealed class Singleton<T> where T: class, new()
    {
        /// <summary>
        /// The singleton instance value.
        /// </summary>
        private static readonly Lazy<T> instance = new Lazy<T>(() => new T());
        
        /// <summary>
        /// The unique, singleton instance.
        /// </summary>                
        public static T Instance
        {
            get { return instance.Value; }
        }

        /// <summary>
        /// A private constructor, this class cannot be constructed.
        /// </summary>
        private Singleton() { }
    } 
```
Then you could have `AssemblyProcessor` as an ordinary class with something like `Singleton<AssemblyProcessor>.Instance.Initialize();`, `Singleton<AssemblyProcessor>.Instance.Dispose();`.

Aah, I'm writing code. I'm so excited! :dancer: